toolbox/repo/validate_role_vars_used.py

```diff
@@ -0,0 +1,78 @@
+#! /usr/bin/env python
+
+import glob, os
+import yaml
+import subprocess
+import pathlib
+
+THIS_DIR = pathlib.Path(__file__).absolute().parent
+
+# go to top directory
+os.chdir(os.path.join(THIS_DIR, "..", ".."))
+
+ROLES_VARS_GLOB = "roles/*/vars/*/*"
+ROLES_DEFAULTS_GLOB = "roles/*/defaults/*/*"
+
+def validate_role_vars_used(filename, yaml_doc):
+    errors = 0
+
+    role_name = filename.split("/", 2)[1]
+    for key in yaml_doc.keys():
+        proc = subprocess.run(["grep", key, f"roles/{role_name}",
+                               "--recursive",
+                               ], capture_output=True)
+
+        if proc.returncode != 0:
+            # grep should always find 'key' inside 'filename'
+            # I couldn't find how to exclude one this file with grep options.
+            print(proc.stderr)
+            print("ERROR: grep shouldn't have failed ...")
+            proc.check_returncode() # raises a subprocess.CalledProcessError
+
+        count = proc.stdout.count(b'\n')
```

Comment:
nit: 
```python
count = len(proc.stdout.decode('utf-8').splitlines())
```
for better unicode support (obviously not important here, it's a nit)

Reply:
I hesitated between binary and utf-8, I thought "if the string can't be converted to utf8 it will fail",
but better assume that utf8 always works, and `b'\n'` might be less generic